app/routes.php

```diff
@@ -0,0 +1,171 @@
+        $settings = callAuthenticationWith($settings, $payload);
+
+        $_SESSION['settings'] = $settings;
+        $response = $container->render($response, 'main.phtml',  ["settings" => $settings]);
+        return $response;
+    });
+
+    //call-api
+    $app->post('/call-api', function (Request $request, Response $response) {
+        $container = new PhpRenderer('../templates/');
+        $settings = $_SESSION['settings'];
+
+        $parsedBody = $request->getParsedBody();
+        $url = filter_var($parsedBody['url'], FILTER_SANITIZE_STRING);
+        $token = filter_var($parsedBody['token'], FILTER_SANITIZE_STRING);
+        $client = new Client();
+        $res = $client->request('GET', $url, ['verify' => false,
+            'headers' => [
+                'authorization' => 'Bearer ' . $token,
+                'Accept' => 'application/vnd.deere.axiom.v3+json'
+            ]
+        ]);
+        $body = (string)$res->getBody();
+        $body = json_encode(json_decode($body), JSON_PRETTY_PRINT);
+
+        $settings['accessToken'] = $token;
+        $settings['apiResponse'] = $body;
+        $_SESSION['settings'] = $settings;
+        $response = $container->render($response, 'main.phtml',  ["settings" => $settings]);
+        return $response;
+    });
+
+    /**
+     * @param $settings
+     * @param array $payload
+     * @return mixed
+     */
+    function callAuthenticationWith($settings, array $payload)
+    {
+        $token_endpoint = get_location_from_metadata($settings['wellKnown'], 'token_endpoint');
+
+        $client = new Client();
+        $basic_auth_header = base64_encode($settings['clientId'] . ':' . $settings['clientSecret']);
+        $res = $client->request('POST', $token_endpoint, ['verify' => false,
+            'headers' => [
+                'authorization' => 'Basic ' . $basic_auth_header,
+                'Accept' => 'application/json',
+                'Content-Type' => 'application/x-www-form-urlencoded'
+            ],
+            'form_params' => $payload
+        ]);
+        $body = (string)$res->getBody();
+        $json = json_decode($body);
+        $settings['accessToken'] = $json->access_token;
+        $settings['refreshToken'] = $json->refresh_token;
+        $settings['accessTokenDetails'] = getAccessTokenDetails($json->access_token);
+        $now = new DateTime();
+        $settings['exp'] = $now->modify("+" . $json->expires_in / (1000 * 60 * 60) . " hour")->format('Y-m-d H:i:s');
+        return $settings;
+    }
+
+    function getAccessTokenDetails($jwt_access_token) {
+        $separator = '.';
+        list($header, $payload, $signature) = explode($separator, $jwt_access_token);
+        return json_encode(json_decode(base64_decode($payload)), JSON_PRETTY_PRINT);
+    }
+
+    /**
+     * @param string|null $wellKnownUrl
+     * @param string|null $element
+     * @return string|null
+     */
+    function get_location_from_metadata(?string $wellKnownUrl, string $element): string
```

Comment:
You have a mix of camelCase and snake_case function names in this class. Please standardize on one naming convention across the entire project.